docs/site/api.md

```diff
@@ -0,0 +1,541 @@
+# About
+
+The DMOJ supports a simple JSON API for accessing most data used by the backend. Currently, there are two versions.
+
+ - [API v2](?id=dmoj-api-v2) - A more rich API that returns user-specific data. This version fully utilizes API tokens.
+ - [API v1](?id=dmoj-api-v1) - Most data returned here are accessible by any user. This version may be deprecated in the future.
+
+## API Tokens
+
+The DMOJ supports API tokens for accessing the majority of the site as your native user, in addition to both API versions. The admin portion of the site is left intentionally innacessible with these tokens. You may generate an API token on your [Edit Profile](https://dmoj.ca/edit/profile) page. To use, include the following header with every request where <API Token> is your API token:
+
+```
+Authorization: Bearer <API Token>
+```
+
+### Error Responses
+The following error codes may be returned by the API token authentication layer. Note that the site itself may return other codes not listed here or identical codes with different error messages, so read the error messages carefully.
+
+ - `400 Invalid authorization header` - The **header** you provided is invalid. Make sure it matches the following regex: `Authorization: Bearer ([a-zA-Z0-9_-]{48})`
+ - `401 Invalid token` - The **token** you provided is invalid. Make sure it matches the one on your [Edit Profile](https://dmoj.ca/edit/profile) page.
+ - `403 Admin inaccessible` - You are trying to access the inaccessible admin portion of the site.
+
+## Rate limiting
```

Comment:
We should mention that this is a prod only feature, and not available in the open source repo.

Reply:
Noted.